### DISCUSSION TITLE
mass media player entities

### DISCUSSION THREAD

**Discussion by johanneswilkens** [music-assistant/support#449] [Q&A]
I was just  wondering why the addon creates its own entities for all the media players, sometimes the mass one is unavailible while the original is online or the other way around. I noticed that its also possible to disable the mass entity creation during setup.
Can someone explain the reason for that and what are the pros and cons of using mass entities over normal media player entities.

**Comment by marcelveldt** [comment]
The mass entities are created because MA adds features on top of the "source" entity, such as queuing support.
Besides that the metadata of what's playing will be known only by MASS (as that handles the queue). There are some exceptions though, for example Linkplay where the author of the Linkplay integration decided to work with me to get his media player native support for the Music Assistant.

If a mass entity is sometimes unavailable while the original is not is considered a bug and not a feature.
In fact, I'm currently working on an improvement that might actually solve that. So please test with the next 2022.6.3 patch release and if you still experience that issue, file an issue report hete in Github.

**Comment by OzGav** [comment]
V1 question
